src/main/java/org/jenkinsci/plugins/docker/commons/fingerprint/DockerFingerprints.java

```diff
@@ -79,9 +80,8 @@ private DockerFingerprints() {} // no instantiation
      * @return Created fingerprint or null if it is not found
      * @throws IOException Fingerprint loading error
      */
-    public static @CheckForNull Fingerprint of(@Nonnull String id) throws IOException {
-        final Jenkins jenkins = Jenkins.getInstance(); // should be not null
-        return jenkins != null ? jenkins.getFingerprintMap().get(getFingerprintHash(id)) : null;
+    public static Fingerprint of(@Nonnull String id) throws IOException {
```

Comment:
You need to leave `@CheckForNull` on this method—there may be no such fingerprint. (I think you got confused by removing the code path where `Jenkins.instance == null`.)

Reply:
Yeah fixed.  Didn't look closely enough at the rest of the code.